include/deal.II/distributed/fully_distributed_tria.h

```diff
@@ -0,0 +1,406 @@
+     * A distributed triangulation with a distributed coarse grid.
+     *
+     * The motivation for parallel::fullydistributed::Triangulation has its
+     * origins in the following observations about complex geometries and/or
+     * about given meshes created by an external mesh generator. We regard
+     * complex geometries as geometries that can be meshed only with a
+     * non-negligible number of coarse cells (>10,000):
+     * - storing the coarse-grid information on every process is too expensive
+     *   from a memory point of view (as done by
+     *   parallel::distributed::Triangulation). Normally, a process only needs a
+     *   small section of the global triangulation, i.e., a small section of the
+     *   coarse grid such that a partitioning of the coarse grid is indeed
+     *   essential. We call cells needed by a process locally relevant cells.
+     * - the distribution of the active cells - on the finest level - among all
+     *   processes by simply partitioning a space-filling curve might not lead
+     *   to an optimal result for triangulations that originate from large
+     *   coarse grids: e.g. partitions that belong to the same process might
+     *   be discontinuous, leading to increased communication amount (within a
+     *   node and beyond). Graph-based partitioning algorithms might be a sound
+     *   alternative.
+     *
+     * To be able to construct a fully partitioned triangulation that
+     * distributes the coarse grid and gives flexibility regarding partitioning,
+     * the following ingredients are required:
+     * - a locally relevant coarse-grid triangulation
+     *   (vertices, cell definition; including a layer of ghost cells)
+     * - a mapping of the locally relevant coarse-grid triangulation into the
+     *   global coarse-grid triangulation
+     * - information about which cell should be refined as well as information
+     *   regarding the subdomain_id, the level_subdomain_id, manifold_id,
+     *   and boundary_id of each cell.
+     *
+     * The ingredients listed above are bundled in the struct
+     * parallel::fullydistributed::ConstructionData. The user has to fill this
+     * data structure - in a pre-processing step - before actually creating the
+     * triangulation. Predefined functions to create ConstructionData
+     * can be found in the namespace dealii::fullydistributed::Util.
+     *
+     * Once the ConstructionData `construction_data` has been constructed, the
+     * triangulation `tria` can be created by calling
+     * `tria.reinit(construction_data);`.
+     *
+     * @note This triangulation supports: 1D/2D/3D, hanging nodes,
+     *       geometric multigrid, and periodicity.
+     *
+     * @note Currently no modifications of the triangulation is supported after
+     *       it has been created, i.e., this type of mesh does not support
+     *       any form of adaptivity, not even simple global refinements and
+     *       coarsenings.
```

Comment:
Doesn't this contradict the previous statement about hanging nodes and multigrid?

Reply:
You can create a triangulation with hanging nodes and multigrid levels (with `reinit`). However, once it has been created, it cannot be altered anymore, i.e. you cannot coarsen or refine afterwards...

Obviously, my comment above is misleading. Do you have any recommendations how phrase it differently?